hibernate-core/src/test/java/org/hibernate/orm/test/join/JoinWithSingleTableInheritanceTest.java

```diff
@@ -0,0 +1,256 @@
+		scope.inTransaction(
+				s -> {
+					s.persist(new ChildEntityA( 11 ));
+					s.persist(new ChildEntityA( 12 ));
+					s.persist(new ChildEntityB( 21 ));
+					RootOne r1 = new RootOne(1);
+					r1.setSomeOtherId( 11 );
+					s.persist( r1 );
+					RootOne r2 = new RootOne(2);
+					r2.setSomeOtherId( 12 );
+					s.persist( r2 );
+					RootOne r3 = new RootOne(3);
+					r3.setSomeOtherId( 21 );
+					s.persist( r3 );
+				}
+		);
+
+		scope.inTransaction(
+				s -> {
+					List<Object[]> l = s.createSelectionQuery( "select r.id, r.someOtherId from RootOne r join ChildEntityA", Object[].class ).list();
+					assertEquals( 6, l.size() );
+				}
+		);
+	}
+
+	@Test
+	public void testRightJoinOnSingleTableInheritance(SessionFactoryScope scope) {
+		scope.inTransaction(
+				s -> {
+					s.persist(new ChildEntityA( 11 ));
+					s.persist(new ChildEntityA( 12 ));
+					s.persist(new ChildEntityB( 21 ));
+					RootOne r1 = new RootOne(1);
+					r1.setSomeOtherId( 11 );
+					s.persist( r1 );
+					RootOne r2 = new RootOne(2);
+					r2.setSomeOtherId( 11 );
+					s.persist( r2 );
+					RootOne r3 = new RootOne(3);
+					r3.setSomeOtherId( 21 );
+					s.persist( r3 );
+				}
+		);
+
+		scope.inTransaction(
+				s -> {
+					List<Object[]> l = s.createSelectionQuery( "select r.id, r.someOtherId, ce.id, ce.disc_col from RootOne r right join ChildEntityA ce on ce.id = r.someOtherId order by ce.id, r.id", Object[].class ).list();
+					assertEquals( 3, l.size() );
+					Object[] r1 = l.get( 0 );
+					assertEquals( 1, (int) r1[0] );
+					assertEquals( 11, (int) r1[1] );
+					assertEquals( 11, (int) r1[2] );
+					assertEquals( 1, (int) r1[3] );
+					Object[] r2 = l.get( 1 );
+					assertEquals( 2, (int) r2[0] );
+					assertEquals( 11, (int) r2[1] );
+					assertEquals( 11, (int) r2[2] );
+					assertEquals( 1, (int) r2[3] );
+					Object[] r3 = l.get( 2 );
+					assertNull( r3[0] );
+					assertNull( r3[1] );
+					assertEquals( 12, (int) r3[2] );
+					assertEquals( 1, (int) r3[3] );
+				}
+		);
+	}
+
+	@Test
+	@RequiresDialectFeature(feature = DialectFeatureChecks.SupportsFullJoin.class)
+	public void testFullJoinOnSingleTableInheritance(SessionFactoryScope scope) {
+		scope.inTransaction(
+				s -> {
+					s.persist(new ChildEntityA( 11 ));
+					s.persist(new ChildEntityA( 12 ));
+					s.persist(new ChildEntityB( 21 ));
+					s.persist(new ChildEntityB( 22 ));
+					RootOne r1 = new RootOne(1);
+					r1.setSomeOtherId( 11 );
+					s.persist( r1 );
+					RootOne r2 = new RootOne(2);
+					r2.setSomeOtherId( 11 );
+					s.persist( r2 );
+					RootOne r3 = new RootOne(3);
+					r3.setSomeOtherId( 21 );
+					s.persist( r3 );
+					RootOne r4 = new RootOne(4);
+					s.persist( r4 );
+				}
+		);
+
+		scope.inTransaction(
+				s -> {
+					List<Object[]> l = s.createSelectionQuery( "select r.id, r.someOtherId from RootOne r full join ChildEntityA ce order by ce.id, r.id", Object[].class ).list();
```

Comment:
You need a join predicate for full joins

Reply:
I added 1=1, but keeps failing on postgresql with that weird error, so i excluded the full join test for it.

Reply:
Why don't you use the same predicate that you use for other join tests? PG works fine with sane queries ;)